### DISCUSSION TITLE
Is the button to switch default branch in git still appearing for anyone else?

### DISCUSSION THREAD

**Discussion by vathanauplight** [community/community#113437] [Repositories]
### Select Topic Area

Bug

### Body

<img width="1029" alt="Screenshot 2024-03-20 at 9 54 10 AM" src="https://github.com/community/community/assets/114683523/e65da997-66a0-4944-a03c-40da4917a2b6">

I'm unable to switch the default branch.. the switch button seems to be missing. I am the repo admin.

**Comment by nehalp13** [comment]
Under your repository name, click Settings. 
If you cannot see the "Settings" tab, select the dropdown menu, then click Settings. 
Under "Default branch", to the right of the default branch name, click "Switch to another branch" beside "Rename branch". 
Select the branch name from the dropdown list and select "Update".

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @vathanauplight ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. I've gone ahead and moved it for you. Good luck!

**Comment by timmyichen** [comment]
This is missing for me too in one of the repos I have access to as an admin as well.

**Comment by prasadmiriyala** [comment]
Facing the same issue, How to get rid of it. Looking for the solution.

**Comment by bugzbuks** [comment]
Same issue, no option to change default branch and I've got full admin rights

**Comment by StephanBezoenNPO** [comment]
Same issue. The default branch has a ruleset applied to it by the organization, does that have any influence here?

**Reply by flitzie** [reply]
I ran into the same issue. I did some tests with a colleague and I noticed that as soon as a repo has got an **organization branch ruleset** applied (!) and in this org. ruleset is defined:

1. "Target branches" either "Default" or "any-other-branch" (the branch pattern does not need to match the current default of the repo) and
2. "Branch rules" either "Restrict deletions" or "Require a pull request before merging" or "Block force pushes"

then both buttons "Rename branch" and "Switch to another branch" disappear. Adding a team to the "Bypass list" where you are part of and this team has got repo admin permissions does not solve the issue.

To suppress renaming or switching the default branch makes sense, if you have org. rulesets with target "Default" only to prevent that admins can bypass org. rules temporarily by switching the branches back and forth, but when you list all the target branches ("master", "main", "release/**/*" etc.) without the default then it should not prevent switching between them.

So for me this looks like a bug.

**Reply by flitzie** [reply]
A little update: When creating new org. rulesets with "Target branches" either "Default" or "any-branch-of-the-current-repo-default" and selecting any kind of "Branch rules" then the buttons will disappear.
If you have a very old org. ruleset without a "Default" branch setting and the repo admin can still not switch from "a-feature-branch-name" to a proper "main" branch then put the rule into "Evaluate" enforcement state for bit of time and then put the status back to "Active". Or just recreate the org. rule.
